ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/test/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2Test.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2021 Ubique Innovation AG <https://www.ubique.ch>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+package ch.admin.bag.covidcertificate.backend.verifier.ws.controller;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import ch.admin.bag.covidcertificate.backend.verifier.ws.util.TestHelper;
+import ch.admin.bag.covidcertificate.backend.verifier.ws.utils.EtagUtil;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.Map;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.http.HttpHeaders;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+public abstract class VerificationRulesControllerV2Test extends BaseControllerTest {
+
+    protected MediaType acceptMediaType;
+
+    private String verificationRulesUrl = "/trust/v2/verificationRules";
+
+    @Value("${testing.disabledModes}")
+    private String[] disabledModes;
+
+    private static final String PATH_TO_VERIFICATION_RULES = "classpath:verificationRulesV2.json";
+    @Test
+    public void verificationRulesTest() throws Exception {
+        // get verification rules
+        MockHttpServletResponse response =
+                mockMvc.perform(get(verificationRulesUrl).accept(acceptMediaType))
+                        .andExpect(status().is2xxSuccessful())
+                        .andReturn()
+                        .getResponse();
+
+        // verify response
+        assertNotNull(response);
+        ObjectNode rules = objectMapper.valueToTree(testHelper.verifyAndReadValue(
+                        response, acceptMediaType, TestHelper.PATH_TO_CA_PEM, Map.class));
+        ObjectNode expected =
+                (ObjectNode) testHelper
+                        .getObjectMapper()
+                        .readTree(
+                                new ClassPathResource("verificationRulesV2.json").getInputStream());
+
+        if(isNoModeRemovalTest()){
+            assert(disabledModes.length == 0);
+        } else {
+            for (String disabledMode : disabledModes) {
+                boolean hasMode = false;
```

Comment:
shouldn't that be something like `modeWasRemoved`?

Reply:
No, it actually checks whether the JSON has the mode to begin with. Otherwise we might test this with a mode that isn't there in the first place, rendering the test moot.

Reply:
ah now I see, could you add some comments explaining what `hasMode` means and why we iterate here for expected?